client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+type TyperMode = 'TYPING' | 'DELETING';
+
+type AllTyperOptions = {
+	delayBetweenCharacters: number;
+	delayBetweenWords: number;
+};
```

Comment:
I'd expect an interface for this. Simple `Options` name is fine for this local type. We'll leave the properties non-optional which works nicely with other mapped types.

```tsx
interface Options {
	delayBetweenCharacters: number;
	delayBetweenWords: number;
};
```

I've also really liked documenting the properties right in the interface:

```suggestion
interface Options {
	/**
	 * The typing delay time in ms between two characters.
	 *
	 * @default 120
	 */
	delayBetweenCharacters: number;
	
	/**
	 * The typing delay in ms between two words.
	 *
	 * @default 1200
	 */
	delayBetweenWords: number;
};
```